integration_test/test_xgboost_run.py

```diff
@@ -146,6 +149,12 @@ def _verify_metric_logging(self, run):
 
     assert run.values['Training time'].value > 0
 
+    if 'early_stopping_rounds' in self.run_params:
+      assert run.values['num_boost_round_before_stopping']
+      if run.values['num_boost_round_before_stopping'].value < run.assignments['num_boost_round']:
```

Comment:
Maybe the `<` should be `<=`

Reply:
I want to check if early stopping actually happened. Thus why I used <; if num_boost_round_before_stopping = num_boost_round then early stopping didn't do anything.

Reply:
I think for some datasets, the train will stop before `early stop` takes effect.

Reply:
then you should also check the else statement, in which case there is no `best_iteration` metric output.